File: netDxf/IO/DxfHatchSplineData.cs
// Copyright (c) netDxf contributors. Licensed under the MIT License.
using netDxf.Blocks;
using netDxf.Entities;

namespace netDxf.IO
{
    internal partial class DxfWriter
    {
        private void ValidateHatchSplineData()
        {
            // Registered block definitions are all exported, including unreferenced
            // and paper-space blocks. Validate before preprocessing or stream writes.
            foreach (Block block in this.doc.Blocks)
                foreach (EntityObject entity in block.Entities)
                    if (entity is Hatch hatch)
                        foreach (HatchBoundaryPath path in hatch.BoundaryPaths)
                            foreach (HatchBoundaryPath.Edge edge in path.Edges)
                                if (edge is HatchBoundaryPath.Spline spline)
                                    HatchSplineData.Validate(spline);
        }
    }
}

File: javascript/netDxf/IO/DxfHatchSplineData.js
// Copyright (c) Daniel Carvajal and netDxf contributors. MIT License; see package LICENSE.
import { Hatch } from '../Entities/Hatch.js';
import { HatchBoundaryPath } from '../Entities/HatchBoundaryPath.js';
import { HatchSplineData } from '../Entities/HatchSplineData.js';
export function ValidateHatchSplineData(document) {
  for (const block of document.Blocks) for (const entity of block.Entities)
    if (entity instanceof Hatch) for (const path of entity.BoundaryPaths) for (const edge of path.Edges)
      if (edge instanceof HatchBoundaryPath.Spline) HatchSplineData.Validate(edge);
}
